src/annotator/integrations/html-side-by-side.js

```diff
@@ -97,14 +97,27 @@ function textRect(text, start = 0, end = text.data.length) {
   return textRectRange.getBoundingClientRect();
 }
 
+/** @param {Element} element */
+function hasFixedPosition(element) {
+  switch (getComputedStyle(element).position) {
+    case 'fixed':
+    case 'sticky':
+      return true;
+    default:
+      return false;
+  }
+}
+
 /**
  * Yield all the text node descendants of `root` that intersect `rect`.
  *
  * @param {Element} root
  * @param {DOMRect} rect
+ * @param {(el: Element) => boolean} shouldVisit - Optional filter that determines
+ *   whether to visit a subtree
  * @return {Generator<Text>}
  */
-function* textNodesInRect(root, rect) {
+function* textNodesInRect(root, rect, shouldVisit = () => true) {
```

Comment:
If you have a parameter of type `(arg1: SomeType, arg2: someType, ...) => ReturnType` it is legal to pass a callback which takes fewer arguments, since the callback will just ignore them. This design is necessary for things like `array.map` since the user will often pass a callback that uses just the item, even though `map` also passes a bunch of additional arguments.